include/flatcc/portable/pdiagnostic.h

```diff
@@ -20,12 +20,22 @@
 
 #if defined(_MSC_VER) && !defined(PDIAGNOSTIC_AWARE_MSVC)
 #define PDIAGNOSTIC_AWARE_MSVC 1
-#elif defined(__clang__) && !defined(PDIAGNOSTIC_AWARE_CLANG)
+#elif !defined(PDIAGNOSTIC_AWARE_MSVC)
+#define PDIAGNOSTIC_AWARE_MSVC 0
+#endif
+
+#if defined(__clang__) && !defined(PDIAGNOSTIC_AWARE_CLANG)
 #define PDIAGNOSTIC_AWARE_CLANG 1
+#elif !defined(PDIAGNOSTIC_AWARE_CLANG)
+#define PDIAGNOSTIC_AWARE_CLANG 0
+#endif
+
 /* Can disable some warnings even if push is not available (gcc-4.2 vs gcc-4.7) */
-#elif ((__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2)) && \
+#if ((__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 2)) && \
```

Comment:
This doesn't check if __GNUC__ is defined before using it. From memory, this is OK as long as it is not in an elif branch (for reasons beyond my comprehension), but we probably shouldn't rely on it if the goal is to clean this up.

Reply:
It's actually not OK here, good catch. The reason it may not be caught is actually a different can of worms: clang defines __GNUC__.  The fact that it also checks the value of these flags will make it a bit uglier but I'll fix it.